tasks/discover-granules/index.js

```diff
@@ -203,6 +203,7 @@ const checkGranuleHasNoDuplicate = async (granuleId, duplicateHandling) => {
   try {
     response = await granules.getGranuleResponse({
       prefix: process.env.STACKNAME,
+      expectedStatusCodes: ['skip', 'error'].includes(duplicateHandling) ? [200, 404] : 200,
```

Comment:
Also.... what happens here in the case of duplicatHandling set to replace?

In that case, it throws on 404 and (due to the api call) retries a few times, and then returns granuleId on 404.

Is there a good reason to have this logic at all in this case, given if we make 404 an expected response in all cases? If no, we should remove the error catch entirely and handle the 404 the same for all responses rather than this divergent logic.

Reply:
So to be clear `duplicateHandling = version` can stand in for <other values of duplicate handling> in this context - if we have it undefined, *or* tell it to version (or add some OTHER possible value) , the way we handle `404` seems like it is the same between `['skip', 'error'].includes(duplicateHandling)` and `!['skip', 'error'].includes(duplicateHandling)`.  

If so, I think we can just set the expected codes to always be `[404, 200]` and handle 404 outside the error block.

Reply:
I'm still not clear on why we need: 

```
    if (responseError.statusCode === 404) {
      return granuleId;
}
```

What is the condition that granules.getGranuleResponse will throw an error such that`responseError.statusCode === 404` given `404` is an expected status code? 

> DiscoverGranulesDuplicateHandlingSkipSpec.js will fail as the 404 will not enter the retry block as it is expectedStatusCode.

How does the spec fail?         If it does, doesn't that imply that there is something that's relying on the retry behavior we're eliminating here?

Reply:
This was discussed in a huddle, and is addressed by [5a208b9](https://github.com/nasa/cumulus/pull/3298/commits/5a208b91293f651f79ecbbd39f01f429b19b32fa)